docs/source/tutorial/basic.rst

```diff
@@ -2,21 +2,586 @@
+
+    The first commands sets up the computer with the following options:
+
+    * *label* (``-L``): tutor
+    * *hostname* (``-H``): localhost
+    * *transport* (``-T``): local
+    * *scheduler* (``-S``): direct
+    * *work-dir* (``-w``): The ``work`` subdirectory of the current directory
+
+    The second command *configures* the computer with a minimum interval between connections (``--safe-interval``) of 5 seconds.
+    For both commands, the *non-interactive* option (``-n``) is added to not prompt for extra input.
+
+    Next, let's set up the code we're going to use for the tutorial:
+
+    .. code-block:: console
+
+        $ verdi code setup -L add --on-computer --computer=tutor -P arithmetic.add --remote-abs-path=/bin/bash -n
+
+    This command sets up a code with *label* ``add`` on the *computer* ``tutor``, using the *plugin* ``arithmetic.add``.
+
+.. note::
+    A typical real-world example of a computer is a remote supercomputing facility.
+    Codes can be anything from a Python script to powerful *ab initio* codes such as Quantum Espresso or machine learning tools like Tensorflow.
+
+Let's have a look at the codes that are available to us:
+
+.. code:: console
+
+    $ verdi code list
+
+You can see a single code ``add@tutor``, with PK = 5, in the printed list.
+This code allows us to add two integers together.
+The ``add@tutor`` identifier indicates that the code with label ``add`` is run on the computer with label ``tutor``.
+To see more details about the computer, you can use the following ``verdi`` command:
+
+.. code:: console
+
+    $ verdi computer show tutor
+
+Note that the *Work directory* has been set up as the ``work`` subdirectory of the current directory.
+This is the directory in which the calculations running on the ``tutor`` computer will be executed.
+
+Let's now start up the ``verdi shell`` again and load the ``add@tutor`` code using its label:
+
+.. code-block:: ipython
 
-See `issue #3981 <https://github.com/aiidateam/aiida-core/issues/3981>`_.
+    In [1]: code = load_code(label='add')
 
+Every code has a convenient tool for setting up the required input, called the builder.
+It can be obtained by using the ``get_builder`` method:
+
+.. code-block:: ipython
+
+    In [2]: builder = code.get_builder()
+
+Using the builder, you can easily set up the calculation by directly providing the input arguments.
+Let's use the ``Int`` node that was created by our previous ``calcfunction`` as one of the inputs:
+
+.. code-block:: ipython
+
+    In [3]: builder.x = load_node(pk=4)
+       ...: builder.y = Int(5)
+
+.. note::
+
+    One nifty feature of the builder is the ability to use tab completion for the inputs.
+    Try it out by typing ``builder.`` + ``<TAB>`` in the verdi shell.
+
+To execute the ``CalcJob``, we use the ``run`` function provided by the AiiDA engine:
+
+.. code-block:: ipython
+
+    In [4]: from aiida.engine import run
+       ...: run(builder)
```

Comment:
This command failed for me with:
```
ValueError: Error occurred validating port 'inputs.metadata.options.resources': required value was not provided for 'resources'
```

Reply:
This is because the changes I've made to the `ArithmeticAddCalculation` calculation job aren't implemented in this branch.

I'll be doing a PR for the demo processes into `develop` once we're entirely settled on them, and the tests have been updated. However, apparently implementing a test for a calculation job is not a trivial task, so @sphuber is going to help me with that. ^^

Reply:
Ok, sounds good. However, we will have to run a final test on a released AiiDA version before we can publish the revised documentation.